test/integration/master/audit_test.go

```diff
@@ -258,24 +243,60 @@ func testAudit(t *testing.T, version string, level auditinternal.Level, enableMu
 		t.Fatalf("Unexpected error: %v", err)
 	}
 
-	if enableMutatingWebhook {
-		if err := createV1beta1MutationWebhook(kubeclient, url+"/mutation"); err != nil {
-			t.Fatal(err)
-		}
+	if err := createV1beta1MutationWebhook(kubeclient, url+"/mutation"); err != nil {
+		t.Fatal(err)
+	}
+
+	tcs := []struct {
+		auditLevel            auditinternal.Level
+		enableMutatingWebhook bool
+		namespace             string
+	}{
+		{
+			auditLevel:            auditinternal.LevelRequestResponse,
+			enableMutatingWebhook: false,
+			namespace:             nonAdmissionWebhookNamespace,
+		},
+		{
+			auditLevel:            auditinternal.LevelMetadata,
+			enableMutatingWebhook: true,
+			namespace:             "webhook-audit-metadata",
+		},
+		{
+			auditLevel:            auditinternal.LevelRequest,
+			enableMutatingWebhook: true,
+			namespace:             "webhook-audit-request",
+		},
+		{
+			auditLevel:            auditinternal.LevelRequestResponse,
+			enableMutatingWebhook: true,
+			namespace:             "webhook-audit-response",
+		},
 	}
 
+	for _, tc := range tcs {
+		t.Run(fmt.Sprintf("%s.%s.%t", version, tc.auditLevel, tc.enableMutatingWebhook), func(t *testing.T) {
+			testAudit(t, version, tc.auditLevel, tc.enableMutatingWebhook, tc.namespace, kubeclient, logFile)
+		})
+	}
+}
+
+func testAudit(t *testing.T, version string, level auditinternal.Level, enableMutatingWebhook bool, namespace string, kubeclient kubernetes.Interface, logFile *os.File) {
 	var lastMissingReport string
+	createNamespace(t, kubeclient, namespace)
+	defer deleteNamespace(t, kubeclient, namespace)
```

Comment:
I don't think you need to delete the namespace if we're only testing each one one time. Issuing a delete for a namespace will only mark it terminating, it will not actually remove the namespace if the namespace cleanup controller is not running.

Reply:
That is good to know. :+1: